src/wallet/wallet2.cpp

```diff
@@ -3916,17 +3924,25 @@ void wallet2::change_password(const std::string &filename, const epee::wipeable_
  * \param keys_file_name Name of wallet file
  * \param password       Password of wallet file
  */
-bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_string& password)
+bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_string& password, const std::string& keys_buf)
 {
+  // determine if loading from file system or string buffer
+  bool use_fs = keys_file_name != ".keys";
```

Comment:
Would be better to implement `load_keys` overload accepting a buffer as an argument.